nomad/core_sched.go

```diff
@@ -958,9 +1030,20 @@ func (c *CoreScheduler) variablesRekey(eval *structs.Evaluation) error {
 			return err
 		}
 
-		// we've now rotated all this key's variables, so set its state
+		// we've rotated this key's variables, but we need to ensure it hasn't
+		// been used to sign any live workload identities before it's safe to
+		// mark as deprecated
+		inUse, err := c.snap.IsRootKeyMetaInUse(keyMeta.KeyID)
```

Comment:
TODO before we can merge: this is causing a failure in `TestCoreScheduler_VariablesRekey` because we're reusing the snapshot from before we deleted the variables, so the keys always look like they're still in use. Will fix.